lib/ansible/modules/network/nxos/nxos_hsrp.py

```diff
@@ -468,6 +468,8 @@ def main():
             commands.extend(command)
 
     if commands:
+        results['changed'] = True
```

Comment:
When the playbook is executed in `check_mode`, it will not make any changes on the device which is why `changed` is returned as `False`. 

If `commands` are generated, it means that the config will be pushed to the device when not run in check_mode.
If `commands` key in not present in the result, it means that no config would be pushed to device when not run in check_mode.

Here is the documentation for this https://docs.ansible.com/ansible/latest/user_guide/playbooks_checkmode.html

Reply:
Ok, maybe I don't understand something, but why standard `copy` module do the same thing, set `changed` key to `True` even in check mode? https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/files/copy.py#L693

Reply:
Documentation also says `will report what changes they would have made rather than making them`. Currently nxos_hsrp doesn't report any changes in check mode.

Reply:
@subuk RE: "Documentation also says will report what changes they would have made rather than making them", this means what config will be pushed to the device not the `changed` key itself. 
The `commands` key should not be empty unless the operation is idempotent. But `changed` key would remain False.

Reply:
Just imagine the situation. I changed switch configuration in git repository and want to see what real commands ansible will execute. So I execute playbook with `--check` flag. And it shows no changes (output in green color and changed=0 in final report), because `changed` flag always set to `False` in check mode. And no commands, because `commands` array always empty in check mode. But when I run playbook without `--check` flag, it executes commands on real device.

Is this a correct behaviour?

> In nxos_hsrp the commands key is set 
Problem that in check mode module exits before `commands` key is set:
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_hsrp.py#L456-L457

Reply:
Liked I said `commands` key should be set based on the idempotent nature of the task regardless of check_mode being True/False. If your task is idempotent it returns empty list, other returns a list of commands that are supposed to be pushed to the device.  

RE: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_hsrp.py#L456-L457 that's a good catch! Yes that's a bug. Are you still willing to work on it?
Thanks!